src/interface_adapter/friends_list/FriendsListPresenter.java

```diff
@@ -17,7 +24,21 @@ public FriendsListPresenter(ViewManagerModel viewManagerModel,
     }
 
     @Override
-    public void prepareSuccessView(FriendsListOutputData friendsListOutputData) {
+    public void prepareSuccessView(FriendsListOutputData response) {
+
+
+        HashMap<String, String> friendsList = response.getFriends();
+
+        FriendsListState friendsListState = friendsListViewModel.getState();
+
+        friendsListState.setFriendsList(friendsList);
+
+        this.friendsListViewModel.setState(friendsListState);
```

Comment:
The way Melissa constructs FriendsList, she didn't use state, so she basically store the data in the view model